src/EditorFeatures/CSharp/EventHookup/EventHookupCommandHandler_SessionCancellingCommands.cs

```diff
@@ -16,14 +17,14 @@ internal partial class EventHookupCommandHandler :
 
         public bool ExecuteCommand(EscapeKeyCommandArgs args, CommandExecutionContext context)
         {
-            AssertIsForeground();
+            Contract.ThrowIfFalse(_threadingContext.HasMainThread);
```

Comment:
❗ `HasMainThread` is a completely different check (it checks for the existence of a main thread). You want this:

```suggestion
            _threadingContext.ThrowIfNotOnUIThread();
```

Reply:
done.